src/jemalloc.c

```diff
@@ -466,7 +466,9 @@ arena_bind(tsd_t *tsd, unsigned ind, bool internal) {
 		for (unsigned i = 0; i < SC_NBINS; i++) {
 			assert(bin_infos[i].n_shards > 0 &&
 			    bin_infos[i].n_shards <= BIN_SHARDS_MAX);
-			bins->binshard[i] = shard % bin_infos[i].n_shards;
+			assert(BIN_SHARDS_MAX < 1 << (sizeof(uint8_t) * 8));
+			bins->binshard[i] = (uint8_t)(shard %
```

Comment:
nit: change the shard define above so that it's converted only once 
```
uint8_t shard = (uint8_t)atomic_fetch_add_u(...)
```

Reply:
arena->binshard_next is unsigned. There we simply use it as a seed; for all we care it can be named "arena_sharding_random_seed". So casting would be intentionally dropping the higher bits. No preference either way here.